api/v1alpha6/openstackmachine_types.go

```diff
@@ -174,6 +175,16 @@ func (r *OpenStackMachine) SetConditions(conditions clusterv1.Conditions) {
 	r.Status.Conditions = conditions
 }
 
+// SetFailureMessage sets the OpenStackMachine status failure message.
+func (r *OpenStackMachine) SetFailureMessage(err error) {
+	r.Status.FailureMessage = pointer.StringPtr(err.Error())
+}
+
+// SetFailureReason sets the OpenStackMachine status failure reason.
+func (r *OpenStackMachine) SetFailureReason(err errors.MachineStatusError) {
+	r.Status.FailureReason = &err
+}
+
```

Comment:
Ideally these would be combined into a single SetFailure() taking both arguments. I don't think we'd ever want to accidentally set one without the other?

This would be ok as a follow-up if you don't want to update this PR, though.

Reply:
Totally agree to combine both functions to `SetFailure()`.

I copied this from CAPA: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/4dda2debcf94a2f9d6cbe0b862bfcd6296178832/pkg/cloud/scope/machine.go#L176-L184